node/src/components/storage.rs

```diff
@@ -209,6 +209,10 @@ pub struct Storage {
     switch_block_era_id_index: BTreeMap<EraId, BlockHash>,
     /// A map of deploy hashes to hashes of blocks containing them.
     deploy_hash_index: BTreeMap<DeployHash, BlockHash>,
+    /// Whether or not memory deduplication is enabled.
+    enable_mem_deduplication: bool,
+    /// Pool of loaded items.
+    pub deploy_cache: BlobCache<<Deploy as Item>::Id>,
```

Comment:
I'd prefer if we kept the cache private and added whatever `pub(crate)` helpers are required.